TickSpec/ScenarioGen.fs

```diff
@@ -481,11 +481,11 @@ let defineStepMethod
 let defineRunMethod
     (scenarioBuilder:TypeBuilder)
     (providerField:FieldBuilder)
-    (beforeScenarioEvents:MethodInfo seq,
-     afterScenarioEvents:MethodInfo seq,
-     beforeStepEvents:MethodInfo seq,
-     afterStepEvents:MethodInfo seq)
-    (stepMethods:seq<MethodBuilder>) =
+    (beforeScenarioEvents:MethodInfo list,
```

Comment:
I'd argue these should become arrays - more emory efficient and not a linked list so potentially better cache locality
Yes technically they are mutable
Eventually there will be an `iarray` alias and can flip to that

Reply:
I have removed the type from there to make it agnostic as long as it is anything with an iterator.